cli/dcos-spark/submit_builder.go

```diff
@@ -298,8 +299,11 @@ func parseApplicationFile(args *sparkArgs) error {
 }
 
 func cleanUpSubmitArgs(argsStr string, boolVals []*sparkVal) ([]string, []string) {
+
+  // collapse two or more spaces to one.
+	argsCompacted := collapseSpacesPattern.ReplaceAllString(argsStr, " ")
 	// clean up any instances of shell-style escaped newlines: "arg1\\narg2" => "arg1 arg2"
-	argsCleaned := strings.TrimSpace(backslashNewlinePattern.ReplaceAllLiteralString(argsStr, " "))
+	argsCleaned := strings.TrimSpace(backslashNewlinePattern.ReplaceAllLiteralString(argsCompacted, " "))
 	// HACK: spark-submit uses '--arg val' by convention, while kingpin only supports '--arg=val'.
 	//       translate the former into the latter for kingpin to parse.
 	args := strings.Split(argsCleaned, " ")
```

Comment:
Another option would be to split `argsCleaned` using a `Regexp.Split` instead of `strings.Split`:
```go
package main

import (
	"fmt"
	"regexp"
)

func main() {
	argString := "one two  three   four"

	splitter := regexp.MustCompile(`\s+`)
	for _, s := range splitter.Split(argString, -1) {
		fmt.Println(s)
	}
}
```

It would also be nice to add `submit_builder_test.go` with some unit tests for the various cases here.

Reply:
Yes there are multiple option to fix this I guess. I agree we need tests since this is not the only issue found so far (probably will need to address it in a separate PR). @susanxhuynh are you runnign any tests for this internally?

Reply:
As far as I know, there are no additional tests for this kind of behaviour, and adding a simple unit tests here would add great value. If you don't get to it, I will see if I can add something this weekend.

Reply:
@elezar ok I will check what I can add here.

Reply:
The CLI is "tested" by the integration tests, they will fail fast if it's not working. Simple unit tests are fine, but IMO opinion we could just add to the CI tests given that the unit tests would be very simple (akin to getter/setter basically).